### ISSUE TITLE
Buildout fails due to uWSGI package name case mismatch

### ISSUE THREAD

**Issue by hathawsh** [buildout/buildout#395]
The following buildout.cfg fails the first time, then succeeds on subsequent attempts:

```
[buildout]
parts = uWSGI

[uWSGI]
recipe = zc.recipe.egg
```

The effect is the same if uWSGI is lowercase.

The error (with Buildout 2.9.3 on Python 3.5 or 3.4) is:
```
While:
  Installing uWSGI.
  Getting distribution for 'uWSGI'.

An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/buildout.py", line 2123, in main
    getattr(buildout, command)(args)
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/buildout.py", line 796, in install
    installed_files = self[part]._call(recipe.install)
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/buildout.py", line 1553, in _call
    return f()
  File "/home/shane/src/uwsgitest/eggs/zc.recipe.egg-2.0.3-py3.5.egg/zc/recipe/egg/egg.py", line 126, in install
    reqs, ws = self.working_set()
  File "/home/shane/src/uwsgitest/eggs/zc.recipe.egg-2.0.3-py3.5.egg/zc/recipe/egg/egg.py", line 84, in working_set
    allow_hosts=self.allow_hosts)
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/easy_install.py", line 913, in install
    return installer.install(specs, working_set)
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/easy_install.py", line 665, in install
    for dist in self._get_dist(requirement, ws):
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/easy_install.py", line 563, in _get_dist
    dists = [_move_to_eggs_dir_and_compile(dist, self._dest)]
  File "/home/shane/src/uwsgitest/eggs/zc.buildout-2.9.3-py3.5.egg/zc/buildout/easy_install.py", line 1730, in _move_to_eggs_dir_and_compile
    assert newdist is not None  # newloc above is missing our dist?!
AssertionError
```

Buildout installs the egg, but thinks the install failed. When Buildout runs the second time, the uWSGI egg is already installed and Buildout completes successfully.

My guess is the _get_matching_dist_in_location function should be doing a case-insensitive comparison of dist.project_name.

**Comment by icemac** [comment]
I previously saw this error, too. A lower/upper case issue seems reasonable to me, although I did not get this far debugging the issue as it vanishes with the next buildout run.

**Comment by daybarr** [comment]
#385 reports a similar case sensitivity problem with the same assertion firing for django/Django. That issue explicitly calls out Windows, but from the traceback here it looks like this is a 'nix issue too?

**Comment by hathawsh** [comment]
@daybarr Correct, this is on Ubuntu 16.04. Thanks for linking the issues together.

**Comment by tseaver** [comment]
I don't know if it is relevant, but recent setuptools has begun to force package-name-normalization, especially when using the `/simple` index pages on PyPI.

**Comment by hathawsh** [comment]
I can confirm this was fixed by #424.
